web/src/main/java/com/futurewei/alcor/web/restclient/NodeManagerRestClient.java

```diff
@@ -34,8 +38,23 @@
 
     @DurationStatistics
     public NodeInfoJson getNodeInfo(String nodeId) throws Exception {
-        String url = nodeManagerUrl + "/" + nodeId;
-        return getRequest(url, NodeInfoJson.class);
+        String queryParameter = buildQueryParameter("node_name", Collections.singletonList(nodeId));
+        String url = nodeManagerUrl + "?" + queryParameter;
+        NodesWebJson nodesWebJson = getForObject(url, NodesWebJson.class);
+        if (nodesWebJson == null ||
+                nodesWebJson.getNodeInfos() == null ||
+                nodesWebJson.getNodeInfos().size() != 1) {
+            return new NodeInfoJson();
+        }
+
+        return new NodeInfoJson(nodesWebJson.getNodeInfos().get(0));
+    }
+
+    @DurationStatistics
+    public NodesWebJson getNodeInfoBulk(List<String> nodeIds) throws Exception {
+        String queryParameter = buildQueryParameter("node_name", nodeIds);
```

Comment:
"node_name" => "name"

Reply:
But node manager uses node_name:
public List<NodeInfo> getAllNodes(@ApiParam(value = "**node_name**") @RequestParam(required = false) String name,
                                      @ApiParam(value = "node_id") @RequestParam(required = false) String id,
                                      @ApiParam(value = "mac_address") @RequestParam(required = false) String mac_address,
                                      @ApiParam(value = "local_Ip") @RequestParam(required = false) String local_ip) throws ParameterNullOrEmptyException, Exception {

https://github.com/futurewei-cloud/alcor/blob/master/services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/controller/NodeController.java

Reply:
@chenpiaoping Somehow the query parameters that actually worked in the E2E test are 
__name__
__id__
__mac_address__
__local_ip__

Is this expected? @Gzure

Reply:
Yes. name is ok as a parameter.